tests/E2E/AssetBasicTest.php

```diff
@@ -23,20 +24,43 @@ public function setUp()
     }
 
     /**
-     * @vcr e2e_asset_get_all.json
+     * @vcr e2e_asset_get_all_locale_all.json
      */
     public function testGetAll()
     {
+        $query = (new Query())
+            ->setLocale('*');
+        $assets = $this->client->getAssets($query);
+
+        $this->assertInstanceOf(ResourceArray::class, $assets);
+    }
+
+    /**
+     * @vcr e2e_asset_get_all_locale_default.json
+     */
+    public function testGetAllSingleLocale()
+    {
         $assets = $this->client->getAssets();
 
         $this->assertInstanceOf(ResourceArray::class, $assets);
     }
 
     /**
-     * @vcr e2e_asset_get_one.json
+     * @vcr e2e_asset_get_one_locale_all.json
      */
     public function testGetOne()
     {
+        $asset = $this->client->getAsset('nyancat', '*');
+
+        $this->assertInstanceOf(Asset::class, $asset);
+        $this->assertEquals('nyancat', $asset->getId());
+    }
+
+    /**
+     * @vcr e2e_asset_get_one_locale_default.json
+     */
+    public function testGetOneSingleLocale()
+    {
         $asset = $this->client->getAsset('nyancat');
 
         $this->assertInstanceOf(Asset::class, $asset);
```

Comment:
The main difference is that the data structure from the API is different for single locale vs. multi locale - those need to be tested here. The actual access in the `Asset` class is the same, as it get normalised before the class is constructed.